upf/upf_pfcp_api.c

```diff
@@ -596,38 +596,38 @@ validate_teid (u32 teid)
   return ((teid != 0) && (teid != 0xffffffff));
 }
 
-static int
-teid_v4_lookup (upf_main_t * gtm, u32 teid, upf_upip_res_t * res)
+static u32
+teid_v4_lookup_session_index (upf_main_t * gtm, u32 teid, ip4_address_t * ip4)
 {
   clib_bihash_kv_8_8_t kv, value;
   gtpu4_tunnel_key_t key4;
 
-  key4.dst = res->ip4.as_u32;
+  key4.dst = ip4->as_u32;
   key4.teid = teid;
 
   kv.key = key4.as_u64;
 
   if (PREDICT_FALSE
       (clib_bihash_search_8_8 (&gtm->v4_tunnel_by_key, &kv, &value)))
-    return UPF_GTPU_ERROR_NO_SUCH_TUNNEL;
+    return ~0;
 
-  return 0;
+  return value.value & 0xffffffff;
```

Comment:
what's that for? The u64 should be truncated automatically to u32 on return.

Reply:
Removed that `&` part